scripts/upgrade.py

```diff
@@ -0,0 +1,74 @@
+import json
+import os
+import re
+
+if os.path.exists("config.json"):
+    with open("config.json", "r") as f:
+        config = json.load(f)
+    major, minor, build = [int(x) for x in config['version'].split(".")]
+    if build > 13:
+        print("Version number is greater than current release.")
+    elif build == 13:
+        print("No upgrade needed. This is the current release version.")
+    else:
+        pass  # This is for future upgrades
+elif os.path.exists("config.py"):
+    print(f"\nImporting current configuration settings.\n\n"
+          f"Go to http://0.0.0.0:8888 to view new configuration interface.\n"
+          f"Replace 0.0.0.0 with the IP address of the Pi, "
```

Comment:
Can you replace this with something like https://www.geeksforgeeks.org/python-program-find-ip-address/ that simply gets the right IP?

Reply:
I'm SSH'd into my pi and the commands are returning `127.0.0.1` but the Pi's IP address is actually `192.168.1.3`. I think, if it's possible, it would be great to find a way to accomplish this, though.

Reply:
`getent hosts mylittlepi.local | awk '{ print $1 }'` seems to work but it's a shell command. The python command from the link fails because of the last line in `/etc/hosts/ binds the hostname to 127.0.0.1:

```
  GNU nano 3.2                                                                             /etc/hosts                                                                                        

127.0.0.1       localhost
::1             localhost ip6-localhost ip6-loopback
ff02::1         ip6-allnodes
ff02::2         ip6-allrouters

127.0.1.1               mylittlepi
```

Reply:
I can look into this. Any ideas on the best way to remove this line (or prevent it from being written)?

Reply:
Try this @metaMMA 

```bash
pi@tart:~ $ python3
Python 3.7.3 (default, Apr  3 2019, 05:39:12)
[GCC 8.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import socket
>>> socket.gethostbyname(socket.gethostname() + ".local")
'192.168.15.214'
>>>
```

Adding `.local` has it lookup the name used for mDNS and such (or, I think it does anyhow)

Reply:
works perfect. Thanks